todo-frontend-react-typescript/src/App.tsx

```diff
@@ -73,37 +72,41 @@ class App extends React.Component<{}, { todos: TodoData }> {
 
   async initializeFeatureHub() {
-    featureHubRepository.addReadynessListener((readyness) => {
-      console.log('readyness', readyness);
-      if (readyness === Readyness.Ready) {
-        const color = featureHubRepository.getFeatureState('SUBMIT_COLOR_BUTTON').getString();
-        this.setState({todos: this.state.todos.changeColor(color)});
-      }
-    });
+          featureHubRepository.addReadynessListener((readyness) => {
+              if (!initialized) {
+                  console.log('readyness', readyness);
+                  if (readyness === Readyness.Ready) {
+                      initialized = true;
+                      const color = featureHubRepository.getFeatureState('SUBMIT_COLOR_BUTTON').getString();
+                      this.setState({todos: this.state.todos.changeColor(color)});
+                  }
+              }
+          });
 
-    featureHubRepository
-      .addPostLoadNewFeatureStateAvailableListener((_) =>
-                                                   this.setState(
-                                                     {todos: this.state.todos.changeFeaturesUpdated(true)}) );
+     //Using catch & release mechanism
+    // featureHubRepository
+    //   .addPostLoadNewFeatureStateAvailableListener((_) =>
+    //                                                this.setState(
+    //                                                  {todos: this.state.todos.changeFeaturesUpdated(true)}) );
 
-    featureHubRepository.catchAndReleaseMode = true; // don't allow feature updates to come through
+    // featureHubRepository.catchAndReleaseMode = true; // catch feature updates and release later
 
     // load the config from the config json file
     const config = (await globalAxios.request({url: 'featurehub-config.json'})).data as ConfigData;
     // setup the api
     todoApi = new DefaultApi(new Configuration({basePath: config.baseUrl }));
     this._loadInitialData(); // let this happen in background
 
-    // listen for reatures from configured source
+    // listen for features from the specified SDK Url for a given environment
     this.eventSource = new FeatureHubEventSourceClient(config.sdkUrl);
     this.eventSource.init();
 
-    // alternative if you want to ready to incoming feature changes and not wait for a page refresh
-    // featureHubRepository.getFeatureState('SUBMIT_COLOR_BUTTON').addListener((fs: FeatureStateHolder) => {
-    //   this.setState({todos: this.state.todos.changeColor(fs.getString())});
-    // });
+    // react to incoming feature changes in real-time
+    featureHubRepository.getFeatureState('SUBMIT_COLOR_BUTTON').addListener((fs: FeatureStateHolder) => {
+      this.setState({todos: this.state.todos.changeColor(fs.getString())});
+    });
 
-    // featureHubRepository.addAnalyticCollector(new GoogleAnalyticsCollector('UA-1234', '1234-5678-abcd-1234'));
+    featureHubRepository.addAnalyticCollector(new GoogleAnalyticsCollector('UA-1234', '1234-5678-abcd-1234'));
```

Comment:
if you do this, it will spam Google when they run the app. You really want to do that? I think you should warn them.

Reply:
spam in what way? I thought it is for when events are fired?

Reply:
yes, but events ARE logged. when you add or remove a todo, events are fired to this rubbish UA + cid.

Reply:
i'm not sure if you mind however!

Reply:
oh i see what you mean